readme.md

```diff
@@ -10,248 +8,408 @@ See also: https://github.com/ga-wdi-lessons/ruby-oop
+    attr_accessor :name
+    attr_reader :age
+
+    # This is run every time we call Person.new
+    # Think of it as a "constructor method"
+    def initialize(initial_name, initial_age)
+      @name = initial_name
+      @age = initial_age
     end
   end
+  ```
+
+  > In place of the `attr` methods, we could also define getter and setter methods.
+
+</details>
+<br/>
+<details>
+  <summary><strong>Why did we place an <code>@</code> in front of our variables? How else could we have written our variables?</strong></summary>
+  <br/>
+
+  `@` indicates that we are defining an instance variable, which is a value that is accessible anywhere inside a given instance of a class.
+
+  The different types of variables we can use are...
+
+
+  #### Local
+
+  - Written using plain vanilla names (e.g., `current_age`, `favorite_color`).
+  - Limited in scope. Only accessible in the current method.
+  - Used for temporary values in a class.
+
+  #### Instance
+
+  - Start with an `@` (e.g., `@name`).
+  - Used to store attributes for a given instance.
+  - Can be used in any instance method in the class.
+  - Very common in ruby.
+
+  #### Class
+
+  - Start with 2 `@`s (e.g., `@@person_count`).
+  - One copy shared between all instances.
+  - Not very common. Over-use is considered bad practice.
+
+</details>
+<br/>
+<details>
+  <summary><strong>How can we generate an instance of this <code>Person</code> class?</strong></summary>
+
+  ```rb
+  adrian = Person.new("Adrian", 29)
+  ```
+</details>
+<br/>
+<details>
+  <summary><strong>How do we go about accessing/modifying a <code>Person</code> instance's <code>name</code> and <code>age</code> attributes?</strong></summary>
 
-  def demonstrate_getting_name
-    # goal is to get our person's name in the best way possible
-    # GETTERS
-    puts @name
-    puts name
-    puts self.name
+  ```rb
+  adrian.name
+  # => "Adrian"
+  adrian.name = "Dwayne 'The Rock' Johnson"
+  adrian.name
+  # =>"Dwayne 'The Rock' Johnson"
+
+  adrian.age
+  # => 29
+  adrian.age = 1
+  # => undefined method `age='
+  ```
+
+</details>
+
+--------
+
+<details>
+  <summary><strong>Let's give our <code>Person</code> class a <code>say_name</code> method that prints his/her name to the console.</strong></summary>
+
+  ```rb
+  class Person
+    attr_accessor :name
+    attr_reader :age
+
+    def initialize(initial_name, initial_age)
+      @name = initial_name
+      @age = initial_age
+    end
+
+    def say_name
+      puts "Hi, my name is " + @name + "."
+    end
   end
 
-  def demonstrate_setting_hunger_level
-    # goal is to set/update our person's hunger level in the best way possible
-    # SETTERS
-    @hunger_level = -10
-    hunger_level = -10
-    self.hunger_level = -10
+  adrian = Person.new("Adrian", 29)
+  adrian.say_name
+  # => "Hi, my name is Adrian."
+  ```
+
+  > We could also use `self.name` in place of the instance variable `@name`. This is actually preferable. We'll explain why later...
```

Comment:
http://stackoverflow.com/questions/1693243/instance-variable-self-vs `self.` is more reliable